### DISCUSSION TITLE
Validator rewards question

### DISCUSSION THREAD

**Discussion by adaki2004** [0xPolygon/polygon-edge#1129] [General]
# [ Validator rewards ]
## Description
I would like to set up faucet contract(s) and therefore I'd like to redirect the funds from validator rewards to that contract on a polygon edge private blockchain.
I know `premine` flag which works fine for me, but how can I access the rewards of the validators (if they have any?)

## Your environment
- Currently running 4 nodes in docker, minikube environment with 4 nodes
- Latest image on dockerhub  

## Expected behavior
Maybe I have forgotten some flags, or searching for something wrong.
polygon.server.ibft: validator key: addr=0x5f00f9614d1b990770744Aed2E3b4C8eFda77cF8

Through RPC i'm querying the validator key for eth_getBalance and shows me 0, while the premined account has the proper amount of native token.

Is there any flag I need to supply in order to get those rewards ?

This is how I invoke polygon-edge to create genesis:

```
polygon-edge genesis \
            --dir "$GENESIS_PATH" \
            --consensus ibft \
            --ibft-validators-prefix-path data- \
            --premine=XXXXXXXXXXX:100000000000000000000000 \
            --bootnode /dns/node-1/tcp/1478/p2p/$(echo $secrets | jq -r '.[0] | .node_id') \
            --bootnode /dns/node-2/tcp/1478/p2p/$(echo $secrets | jq -r '.[1] | .node_id')
```

And this is how i start the nodes:
```
    - args:
        - server
        - --data-dir
        - /data/data-1
        - --chain
        - /data/genesis.json
        - --grpc-address
        - 0.0.0.0:9632
        - --libp2p
        - 0.0.0.0:1478
        - --jsonrpc
        - 0.0.0.0:8545
        - --prometheus
        - 0.0.0.0:5001
        - --seal
```

Thank you in advance!

**Comment by laviniat1996** [comment]
Hello @adaki2004 ,
The rewards for each validator are stored on that validator's account. To get the rewards, you can use their private keys, import them in a wallet and access those funds.

**Comment by adaki2004** [comment]
Hello @laviniat1996 !

Thanks for the response. I tried to query the result via an RPC call.

Let me summarize the process:
I initiated 4 nodes and premined some eth to a specific address.

This is how I query the results via the rpc endpoint (after redirected to localhost).
```
curl --location --request POST 'localhost:8545' \ 
--header 'Content-Type: application/json' \
--data-raw '{
    "jsonrpc": "2.0",
    "id": 2,
    "method": "eth_getBalance", 
    "params": ["0xD94eEBa4cD11BF55bE43B0B418D9D5739FDEFc40","latest"]
}'
{"jsonrpc":"2.0","id":2,"result":"0x0"}
```

When the address is the node's address the result is 0, when the address is the premined address it is correct. 

This is how I get the node's public address (from the logs):

```
Executing polygon-edge...
2023-01-16T11:41:13.183Z [INFO]  polygon.server: Data dir: path=/data/data-1
2023-01-16T11:41:13.183Z [INFO]  polygon.server: Prometheus server started: addr==0.0.0.0:5001
2023-01-16T11:41:13.208Z [INFO]  polygon.blockchain: genesis: hash=0x295dc6ee0d52fd3421bf550e82310377c8d8d752c895ebda72539c261f28774b
2023-01-16T11:41:13.208Z [INFO]  polygon.server.ibft: validator key: addr=0xD94eEBa4cD11BF55bE43B0B418D9D5739FDEFc40
```

Maybe I forgot a flag.. ? Or why I'm not able to see the rewards distributed to the nodes ?
